lock/src/test/java/org/infinispan/lock/ClusteredLockSplitBrainTest.java

```diff
@@ -1,40 +1,60 @@
 import static org.infinispan.functional.FunctionalTestUtils.await;
+import static org.infinispan.lock.impl.ClusteredLockModuleLifecycle.CLUSTERED_LOCK_CACHE_NAME;
 import static org.infinispan.test.Exceptions.assertException;
 import static org.testng.AssertJUnit.assertFalse;
 import static org.testng.AssertJUnit.assertNotNull;
-import static org.testng.AssertJUnit.assertNull;
 import static org.testng.AssertJUnit.assertTrue;
 import static org.testng.AssertJUnit.fail;
 
+import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionException;
 import java.util.concurrent.TimeUnit;
 
+import org.infinispan.configuration.cache.ConfigurationBuilder;
 import org.infinispan.lock.api.ClusteredLock;
 import org.infinispan.lock.api.ClusteredLockManager;
 import org.infinispan.lock.exception.ClusteredLockException;
 import org.infinispan.manager.EmbeddedCacheManager;
 import org.infinispan.partitionhandling.AvailabilityException;
 import org.infinispan.partitionhandling.BasePartitionHandlingTest;
+import org.infinispan.test.fwk.TransportFlags;
 import org.testng.annotations.Test;
 
 @Test(groups = "functional", testName = "clusteredLock.ClusteredLockSplitBrainTest")
 public class ClusteredLockSplitBrainTest extends BasePartitionHandlingTest {
 
    private static final String LOCK_NAME = "ClusteredLockSplitBrainTest";
 
+   public ClusteredLockSplitBrainTest() {
+      this.numMembersInCluster = 6;
+      //this.cacheMode = CacheMode.REPL_SYNC;
+   }
+
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      ConfigurationBuilder dcc = cacheConfiguration();
+      dcc.clustering().cacheMode(cacheMode).partitionHandling().whenSplit(partitionHandling).mergePolicy(mergePolicy);
+      if (biasAcquisition != null) {
+         dcc.clustering().biasAcquisition(biasAcquisition);
+      }
+      createClusteredCaches(numMembersInCluster, dcc, new TransportFlags().withFD(true).withMerge(true));
+      waitForClusterToForm(CLUSTERED_LOCK_CACHE_NAME);
+   }
+
    @Test
    public void testLockCreationWhenPartitionHappening() throws Throwable {
       ClusteredLockManager clusteredLockManager = EmbeddedClusteredLockManagerFactory.from(getCacheManagers().get(0));
       await(clusteredLockManager.remove(LOCK_NAME));
 
-      splitCluster(new int[]{0, 1}, new int[]{2, 3});
+      splitCluster(new int[]{0, 1, 2}, new int[]{3, 4, 5});
 
-      eventuallyEquals(2, () -> advancedCache(0).getRpcManager().getTransport().getMembers().size());
-      eventuallyEquals(2, () -> advancedCache(1).getRpcManager().getTransport().getMembers().size());
-      eventuallyEquals(2, () -> advancedCache(2).getRpcManager().getTransport().getMembers().size());
-      eventuallyEquals(2, () -> advancedCache(3).getRpcManager().getTransport().getMembers().size());
+      for (int i = 0; i < 6; i++) {
+         int finalI = i;
+         eventuallyEquals(3, () -> advancedCache(finalI).getRpcManager().getTransport().getMembers().size());
```

Comment:
This is basically what `TestingUtil.blockUntilViewReceived(long, Cache...)` does.